vllm/v1/outputs.py

```diff
@@ -12,34 +12,46 @@
 
 
 class LogprobsLists(NamedTuple):
-    # [num_reqs, max_num_logprobs + 1]
+    # [num_reqs x num_generated_tokens, max_num_logprobs + 1]
     logprob_token_ids: list[list[int]]
-    # [num_reqs, max_num_logprobs + 1]
+    # [num_reqs x num_generated_tokens, max_num_logprobs + 1]
     logprobs: list[list[float]]
-    # [num_reqs]
+    # [num_reqs x num_generated_tokens]
     sampled_token_ranks: list[int]
-
-    def slice(self, start: int, end: int):
+    # [num_reqs]
+    cu_num_generated_tokens: Optional[list[int]] = None
```

Comment:
This is needed/used only for slicing right? (may also be worth mentioning that in the comment.
```suggestion
    _cu_num_generated_tokens: Optional[list[int]] = None
```